src/highdicom/pr/content.py

```diff
@@ -737,3 +757,1295 @@ def __init__(
+                raise TypeError(
+                    'Arguments "rescale_intercept", "rescale_slope", and '
+                    '"rescale_type" must either all be provided or none of '
+                    'them shall be provided.'
+                )
+            else:
+                try:
+                    ds = _get_modality_lut_attributes(referenced_images)
+                except (ValueError, AttributeError):
+                    logger.debug(
+                        'no Modality LUT attributes found in referenced images'
+                    )
+                logger.debug(
+                    'use Modality LUT attributes from referenced images'
+                )
+                _add_modality_lut_attributes(
+                    self,
+                    rescale_intercept=ds.RescaleIntercept,
+                    rescale_slope=ds.RescaleSlope,
+                    rescale_type=ds.RescaleType
+                )
+
+        # Softcopy VOI LUT
+        if softcopy_voi_luts is not None:
+            if len(softcopy_voi_luts) == 0:
+                raise ValueError(
+                    'Argument "softcopy_voi_luts" must not be empty.'
+                )
+            for v in softcopy_voi_luts:
+                if not isinstance(v, SoftcopyVOILUT):
+                    raise TypeError(
+                        'Items of "softcopy_voi_luts" must be of type '
+                        'SoftcopyVOILUT.'
+                    )
+
+            if len(softcopy_voi_luts) > 1:
+                if not all(
+                    hasattr(v, 'ReferencedImageSequence')
+                    for v in softcopy_voi_luts
+                ):
+                    raise ValueError(
+                        'If multiple items are passed in "softcopy_voi_luts", '
+                        'each must specify the images that it applies to.'
+                    )
+            _add_softcopy_voi_lut_attributes(
+                self,
+                referenced_images=referenced_images,
+                softcopy_voi_luts=softcopy_voi_luts
+            )
+        else:
+            try:
+                ds = _extract_softcopy_voi_lut_attributes(referenced_images)
+            except (AttributeError, ValueError):
+                logger.debug(
+                    'no VOI LUT attributes found in referenced images'
+                )
+            if len(ds.SoftcopyVOILUTSequence) > 0:
+                logger.debug(
+                    'use VOI LUT attributes from referenced images'
+                )
+                _add_softcopy_voi_lut_attributes(
+                    self,
+                    referenced_images=referenced_images,
+                    softcopy_voi_luts=ds.SoftcopyVOILUTSequence
+                )
+
+        # Palette Color Lookup Table
+        _add_palette_color_lookup_table_attributes(
+            self,
+            palette_color_lut=palette_color_lut
+        )
+
+
+class BlendingDisplayInput(Dataset):
```

Comment:
I generally agree. However, since this sequence attribute exists in the standard, it may be further extended in the future (i.e., additional attributes may be added) and then this logic would break. Therefore, I would prefer keeping it as is, even it leads to slightly more verbose code.